cloud/amazon/route53_facts.py

```diff
@@ -0,0 +1,433 @@
+    params = dict()
+
+    if module.params.get('hosted_zone_id'):
+        params['HostedZoneId'] = module.params.get('hosted_zone_id')
+    else:
+        module.fail_json(msg="Hosted Zone Id is required")
+
+    if module.params.get('start_record_name'):
+        params['StartRecordName'] = module.params.get('start_record_name')
+
+    if module.params.get('type') and not module.params.get('start_record_name'):
+        module.fail_json(msg="start_record_name must be specified if type is set")
+    elif module.params.get('type'):
+        params['StartRecordType'] = module.params.get('type')
+
+    results = client.list_resource_record_sets(**params)
+    return results
+
+
+def health_check_details(client, module):
+    health_check_invocations = {
+        'list': list_health_checks,
+        'details': get_health_check,
+        'status': get_health_check,
+        'failure_reason': get_health_check,
+        'count': get_count,
+        'tags': get_resource_tags,
+    }
+
+    results = health_check_invocations[module.params.get('health_check_method')](client, module)
+    return results
+
+
+def hosted_zone_details(client, module):
+    hosted_zone_invocations = {
+        'details': get_hosted_zone,
+        'list': list_hosted_zones,
+        'list_by_name': list_hosted_zones_by_name,
+        'count': get_count,
+        'tags': get_resource_tags,
+    }
+
+    results = hosted_zone_invocations[module.params.get('hosted_zone_method')](client, module)
+    return results
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        query=dict(choices=[
+            'change',
+            'checker_ip_range',
+            'health_check',
+            'hosted_zone',
+            'record_sets',
+            'reusable_delegation_set',
+        ], required=True),
+        change_id=dict(),
+        hosted_zone_id=dict(),
+        max_items=dict(type='str'),
+        next_marker=dict(),
+        delegation_set_id=dict(),
+        start_record_name=dict(),
+        type=dict(choices=[
+            'A', 'CNAME', 'MX', 'AAAA', 'TXT', 'PTR', 'SRV', 'SPF', 'NS'
+        ]),
+        dns_name=dict(),
+        resource_id=dict(type='list', aliases=['resource_ids']),
+        health_check_id=dict(),
+        hosted_zone_method=dict(choices=[
+            'details',
+            'list',
+            'list_by_name',
+            'count',
+            'tags'
+        ], default='list'),
+        health_check_method=dict(choices=[
+            'list',
+            'details',
+            'status',
+            'failure_reason',
+            'count',
+            'tags',
+        ], default='list'),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        mutually_exclusive=[
+            ['hosted_zone_method', 'health_check_method'],
+        ],
+    )
+
+    # Validate Requirements
+    if not (HAS_BOTO or HAS_BOTO3):
+        module.fail_json(msg='boto3 or boto is required.')
```

Comment:
There is a chance this might fail when there is an ImportError for json. Not that likely but worth considering. Perhaps just change the message to say json and boto/boto3 are required, or handle the json ImportError separately.